optional/vtd/src/vtd_binding.cpp

```diff
@@ -627,19 +627,44 @@ class VtdBinding : public cloe::Simulator {
   }
 
   void apply_scp_scenario_response(boost::property_tree::ptree& xml) {
-    auto trafficcontrol = xml.get_child("Scenario").get_child("TrafficControl");
-    for (auto& it : trafficcontrol) {
-      if (it.first != "Player") continue;
-      auto p = it.second;
-      std::string control = p.get("Description.<xmlattr>.Control", "default");
-      if (control == "external") {
-        auto name = p.get("Description.<xmlattr>.Name", "unspecified");
-        agents_expected_.insert(name);
-
-        // Ask VTD to create a vehicle dynamics instance for this vehicle
-        scp::DynamicsPluginConfig cfg;
-        cfg.name = name;
-        scp_client_->send(cfg);
+    boost::optional<boost::property_tree::ptree&> scenario = xml.get_child_optional("Scenario");
+    if (scenario) {
+      auto trafficcontrol = xml.get_child("Scenario").get_child("TrafficControl");
+      for (auto& it : trafficcontrol) {
+        if (it.first != "Player") continue;
```

Comment:
While we're editing this, lets fix this style to use brackets:
```suggestion
        if (it.first != "Player") {
          continue;
        }
```